src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -164,13 +166,27 @@ private Task StartAsyncCore()
                         _logger.HttpConnectionClosed();
                         return;
                     }
-
-                    _transport = _transportFactory.CreateTransport(GetAvailableServerTransports(negotiationResponse));
+                    _serverTransports = GetAvailableServerTransports(negotiationResponse);
                     connectUrl = CreateConnectUrl(Url, negotiationResponse);
                 }
 
                 _logger.StartingTransport(_transport, connectUrl);
-                await StartTransport(connectUrl);
+                foreach (var transport in _allTransports)
+                {
+                    try
+                    {
+                        if ((transport & _serverTransports) != 0)
```

Comment:
This happens internally in he DefaultTransportFactory https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets.Client.Http/DefaultTransportFactory.cs#L39
Only thing I can think of is if a user passes a custom implementation of ITransportFactory then it might not and we should consider doing the mask ourselves

Reply:
We should actually do the mask here as well. We'll avoid unnecessary calls into `StartTransport` that way